src/megaapi_impl.cpp

```diff
@@ -16985,10 +16982,17 @@ void MegaApiImpl::update()
 int MegaApiImpl::isWaiting()
 {
 #ifdef ENABLE_SYNC
-    if (waiting || client->syncfslockretry)
+    if (client->syncfslockretry || client->syncfsopsfailed)
+    {
+        LOG_debug << "SDK waiting for a blocked file: " << client->blockedfile;
```

Comment:
The SDK will keep retrying on both cases. `syncfslockretry` means that there are files that can't be scanned at the moment, while `syncfsopsfailed` means that there are external changes that can't be applied to the local filesystem at the moment. Both should be temporary problems, usually because the affected file is blocked by another app.